### ISSUE TITLE
Integration with new field that will happen in the install process

### ISSUE THREAD

**Issue by abe-njama** [holochain/holochain#3956]
Depends on #3955 

On install:

- Check the `use_existing` and look for a DNA lineage in other apps that allow the app being installed to run.
- Check for lineage forks and reject the install, require the user/developer to resolve the ambiguity.
- Add a field to `InstallApp` that would allow somebody to resolve an ambiguous lineage and continue with installation without needing a developer to change something.
- Return helpful error messages to explain what is missing so that users can work out what to go and install. Error should be human readable but also ensure that something like Launcher could use the missing DNA hashes to try and figure out what app the user might need to install.

**Comment by maackle** [comment]
More specs:

## Additions
Add new variant `CellProvisioning::UseExisting(Either<DnaHash, CellId>)`. (It does not need a `deferred` flag.)
This is just a recommendation for the type to use, perhaps there is a more convenient one.

## Functionality
UseExisting takes either a DnaHash or a CellId.
If a DnaHash is provided, then the lineages of all DNAs in all currently installed Cells will be checked for the `dna_hash` specified, and:
- If there are zero matches: the installation will fail with an error that there is no cell which can be used
- If there is exactly one match, the installation will proceed using the cell that matched
- If there are more than one match, an error will be returned with the matching CellIds, requiring the user to choose one

A CellId can be provided instead of a DnaHash to disambiguate the last case, or in any situation where the CellId is already known.

The `dna` part of the role manifest can mostly be ignored. `installed_hash` could be used, but this seems pointless, since if you know the hash, you would have provided a CellId with that hash.

**Comment by jost-s** [comment]
Test at least 4 cases:
- existing DNA is clear (= unambiguous), is looked up and app is installed
- existing DNA cannot be found
- existing DNA is ambiguous and returns error message for every multi-version DNA with versions to choose from
- installation provides extra argument for every ambiguous DNA to choose a version, installation succeeds
- - installation provides extra argument for every ambiguous DNA to choose a version, wrong DNA hash cannot be found and installation fails

**Comment by maackle** [comment]
We should actually never automatically select a cell. We need a new admin call that returns the list of cells running a DNA in the lineage of the given DNA. Then the InstallApp call always needs to specify which CellId to use for UseExisting.

**Comment by abe-njama** [comment]
@zippy please take note of this issue that is a dependency to the Dependent DNA story.
